src/dal/interface/index.ts

```diff
@@ -233,18 +233,14 @@ export class DbClient implements IDbClient {
     return result;
   }
 
-  public async getTarget(searchParams: ITargetParams): Promise<IClaim> {
-    const { subject, grades, claimNumber, targetShortCode } = searchParams;
+  public async getTarget(targetShortCode: string): Promise<IClaim> {
     let result: IClaim | undefined;
     if (this.db) {
       try {
         result = (await this.db.collection('claims').findOne({
-          subject,
-          grades,
-          claimNumber,
           'target.shortCode': targetShortCode
         })) as IClaim;
-        result.target = result.target.filter(t => t.shortCode === targetShortCode);
+        // result.target = result.target.filter(t => t.shortCode === targetShortCode);
```

Comment:
Is there some reason why this line is commented out and not just removed?

Reply:
I can definitely not think of a good reason for that, I'll remove it.